bundles/org.openhab.binding.generacmobilelink/src/main/java/org/openhab/binding/generacmobilelink/internal/handler/GeneracMobileLinkAccountHandler.java

```diff
@@ -0,0 +1,267 @@
+            request.send(new BufferingResponseListener() {
+                @NonNullByDefault({})
+                @Override
+                public void onComplete(Result result) {
+                    futureResult.complete(new HTTPResult(result.getResponse().getStatus(), getContentAsString()));
+                }
+            });
+            HTTPResult result = futureResult.get();
+            logger.trace("Response - status: {} content: {}", result.responseCode, result.content);
+            return result;
+        } catch (ExecutionException e) {
+            return new HTTPResult(0, e.getMessage());
+        }
+    }
+
+    private void handleErrorResponse(HTTPResult result) {
+        switch (result.responseCode) {
+            case HttpStatus.UNAUTHORIZED_401:
+                // the server responds with a 500 error in some cases when credentials are not correct
+            case HttpStatus.INTERNAL_SERVER_ERROR_500:
+                // server returned a valid error response
+                ErrorResponseDTO error = gson.fromJson(result.content, ErrorResponseDTO.class);
+                if (error != null && error.errorCode > 0) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                            "Unauthorized: " + result.content);
+                    authToken = null;
+                    break;
+                }
+            default:
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, result.content);
+        }
+    }
+
+    private void updateGeneratorThings() {
+        GeneratorStatusResponseDTO generatorsLocal = generators;
+        if (generatorsLocal != null) {
+            generatorsLocal.forEach(generator -> {
+                Thing thing = getThing().getThing(new ThingUID(GeneracMobileLinkBindingConstants.THING_TYPE_GENERATOR,
+                        getThing().getUID(), String.valueOf(generator.gensetID)));
+                if (thing == null) {
+                    generatorDiscovered(generator);
+                } else {
+                    ThingHandler handler = thing.getHandler();
+                    if (handler != null) {
+                        ((GeneracMobileLinkGeneratorHandler) handler).updateGeneratorStatus(generator);
+                    }
+                }
+            });
+        }
+    }
+
+    private void generatorDiscovered(GeneratorStatusDTO generator) {
+        DiscoveryResult result = DiscoveryResultBuilder
+                .create(new ThingUID(GeneracMobileLinkBindingConstants.THING_TYPE_GENERATOR, getThing().getUID(),
+                        String.valueOf(generator.gensetID)))
+                .withLabel("MobileLink Generator " + generator.generatorName)
+                .withProperty("generatorId", String.valueOf(generator.gensetID))
+                .withRepresentationProperty("generatorId").withBridge(getThing().getUID()).build();
+        discoveryService.thingDiscovered(result);
+    }
```

Comment:
This logic should probably be moved into the GeneracMobileLinkDiscoveryService instead.